src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PointsToAnalysis/PointsToAnalysis.PointsToDataFlowOperationVisitor.cs

```diff
@@ -923,22 +925,38 @@ private PointsToAbstractValue VisitInvocationCommon(IOperation operation, IOpera
                 return value;
             }
 
-            private static bool IsSpecialEmptyOrFactoryMethod(IMethodSymbol method)
-                => IsSpecialFactoryMethod(method) || IsSpecialEmptyMember(method);
+            private static bool IsSpecialMethodReturningNonNullValue(IMethodSymbol method, WellKnownTypeProvider wellKnownTypeProvider)
+                => IsSpecialFactoryMethodReturningNonNullValue(method, wellKnownTypeProvider) || IsSpecialEmptyMember(method);
 
             /// <summary>
             /// Returns true if this special static factory method whose name starts with "Create", such that
+            /// method's return type is not nullable, i.e. 'type?', and
             /// method's containing type is static OR a special type OR derives from or is same as the type of the field/property/method return.
             /// For example: class SomeType { static SomeType CreateXXX(...); }
             /// </summary>
-            private static bool IsSpecialFactoryMethod(IMethodSymbol method)
+            private static bool IsSpecialFactoryMethodReturningNonNullValue(IMethodSymbol method, WellKnownTypeProvider wellKnownTypeProvider)
             {
-                return method.IsStatic &&
+                var isFactoryMethod = method.IsStatic &&
                     method.Name.StartsWith("Create", StringComparison.Ordinal) &&
+                    !method.ReturnType.IsAnnotatedAsNullable() &&
                     (method.ContainingType.IsStatic ||
                      method.ContainingType.SpecialType != SpecialType.None ||
                      method.ReturnType is INamedTypeSymbol namedType &&
                      method.ContainingType.DerivesFromOrImplementsAnyConstructionOf(namedType.OriginalDefinition));
+                if (!isFactoryMethod)
+                {
+                    return false;
+                }
+
+                // Activator.CreateInstance can return 'null'
+                if (method.Name.Equals("CreateInstance", StringComparison.Ordinal) &&
```

Comment:
Is this necessary? The method is correctly annotated.

https://docs.microsoft.com/en-us/dotnet/api/system.activator.createinstance?view=net-5.0#System_Activator_CreateInstance_System_Type_

Reply:
I added it because the VB test was failing.

Reply:
@mavasani In that case, I think we should at least check for `CreateInstanceFrom` as well.

Or, if there is a way to find the correct annotation of C# code (in that case, the runtime) that's being consumed in VB.

Reply:
It's probably good to add a comment here because when looking at the doc we see that the API returns `object?` so it seems weird to have a special case.

Reply:
Thanks, done.